### PR TITLE
Provide assessment assignment

### PR DESCRIPTION
**Pull request by roy-tc**
This PR includes the assessment assignment used by TradeCloud

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,470 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (marcmatt) [inline .idea/workspace.xml]: Remove .idea/workspace.xml, your settings should not be in the repo

Comment (vovinacci) [reply .idea/workspace.xml]: Best idea would be also to add `.idea` to `.gitignore`

Comment (roy-tc) [reply .idea/workspace.xml]: Agreed. This line got kicked out of my .gitignore in the process of initializing the repo.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
 *.class
 *.log
 *.iml
+.idea
+target
```
Comment (vovinacci) [inline .gitignore]: Add `\r` on EOF?

File: Readme.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+### Extended Assignment
+Add a new module to the SBT project that contains a second microservice, which takes care of sending out emails to contacts.
+- Use Kafka for the communication between the microservices. See the section below about how to start Kafka.
+- Your new microservice should send out an email to every newly registered contact using the [`EmailMock`](contacts/src/main/scala/nl/tradecloud/microservicesassignment/email/EmailMock.scala)
+
+## Starting Kafka
+In order to start kafka, running the following commands in the root of the project:
+```bash
+export DOCKER_KAFKA_HOST=$(ipconfig getifaddr en0)
```
Comment (vovinacci) [inline Readme.md]: That works on macOS. Linux way would be `hostname --ip-address`. Should we mention this?

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    image: wurstmeister/kafka
+    ports:
+      - "9092:9092"
+    environment:
+      KAFKA_ADVERTISED_HOST_NAME: ${DOCKER_KAFKA_HOST}
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+    depends_on:
+      - "zookeeper"
```
Comment (vovinacci) [inline docker-compose.yml]: I propose to use slightly different setup for Kafka and Docker compose based on https://hub.docker.com/r/spotify/kafka/
```
version: '2.4'
services:
  kafka:
    image: spotify/kafka:latest
    ports:
      - 9092:9092
      - 2181:2181
    environment:
      - ADVERTISED_HOST=kafka
      - ADVERTISED_PORT=9092
    mem_limit: 1G
```
And also, it's nice to have `\r` at EOF :)

File: contacts/src/main/resources/swagger/swagger.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        example: "Roy de Bokx"
+      telephone:
+        type: string
+        example: "061234567"
+    required:
+      - email
+      - name
+      - telephone
+  Email:
+    type: string
```
Comment (vovinacci) [inline contacts/src/main/resources/swagger/swagger.yml]: `\r` on EOF?

File: contacts/src/main/scala/nl/tradecloud/microservicesassignment/email/EmailMock.scala [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package nl.tradecloud.microserviceassignment.email
+
+class EmailMock() {
+
+  def sendEmail(email: String) = {
+    println(s"Welcome email was sent to ${email}")
+  }
+}
```
Comment (vovinacci) [inline contacts/src/main/scala/nl/tradecloud/microservicesassignment/email/EmailMock.scala]: `\r` on EOF
If Intellij IDEA being used, it's a setting that should be turned on
Preferences > Editor > General > Ensure line feed at file end on save

File: docker-compose.yml (hunk 4 above)
Comment (roy-tc) [reply docker-compose.yml]: Good suggestion. I've changed the image it's using.

File: Readme.md (hunk 3 above)
Comment (roy-tc) [reply Readme.md]: Not a bad idea either. I've updated the readme

**Pull request merged**